src/main/java/pl/asie/computronics/integration/gregtech/gregtech5/DriverTTMultiblockBase.java

```diff
@@ -0,0 +1,55 @@
+package pl.asie.computronics.integration.gregtech.gregtech5;
+
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.world.World;
+import net.minecraftforge.common.util.ForgeDirection;
+
+import gregtech.api.metatileentity.BaseMetaTileEntity;
+import li.cil.oc.api.machine.Arguments;
+import li.cil.oc.api.machine.Callback;
+import li.cil.oc.api.machine.Context;
+import li.cil.oc.api.network.ManagedEnvironment;
+import li.cil.oc.api.prefab.DriverSidedTileEntity;
+import pl.asie.computronics.integration.ManagedEnvironmentOCTile;
+import tectech.thing.metaTileEntity.multi.base.Parameters;
+import tectech.thing.metaTileEntity.multi.base.TTMultiblockBase;
+
+public class DriverTTMultiblockBase extends DriverSidedTileEntity {
+
+    public static class ManagedEnvironmentTTMultiblockBase extends ManagedEnvironmentOCTile<BaseMetaTileEntity> {
+
+        public ManagedEnvironmentTTMultiblockBase(BaseMetaTileEntity tile, String name) {
+            super(tile, name);
+        }
+
+        @Callback(doc = "function(hatch:int, id: int, val:double); Sets a parameter", direct = true)
+        public Object[] setParameters(Context c, Arguments a) {
+            Parameters p = ((TTMultiblockBase) tile.getMetaTileEntity()).parametrization;
+            p.trySetParameters(a.checkInteger(0), a.checkInteger(1), a.checkDouble(2));
+            return null;
+        }
+
+        @Callback(doc = "function(hatch:int, id: int);double Returns the value of a parameter", direct = true)
```

Comment:
Same spacing as above (direct should be fine here though)

Reply:
u got it boss